inmemorystorage/storage.py

```diff
@@ -91,11 +91,12 @@ def __init__(self, dirs=None, files=None, parent=None):
         self.parent = parent
 
     def resolve(self, path, create=False, use_bytes=False):
-        path_bits = path.strip('/').split('/', 1)
+        path = os.path.normpath(path)
+        path_bits = path.strip(os.sep).split(os.sep, 1)
         current = path_bits[0]
         rest = path_bits[1] if len(path_bits) > 1 else None
         if not rest:
-            if current == '':
+            if current == '.' or current == '':
```

Comment:
The check against `.` is needed because it is the result of `normpath('')`. The blank string is needed because a `path` of `/` will still give you a blank string for both `current` and `rest`.